clients/drcachesim/drptpostprocessor/drptpostprocessor.c

```diff
@@ -0,0 +1,649 @@
+/* **********************************************************
+ * Copyright (c) 2022 Google, Inc.  All rights reserved.
+ * **********************************************************/
+
+/*
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of Google, Inc. nor the names of its contributors may be
+ *   used to endorse or promote products derived from this software without
+ *   specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/* DrPTPostprocessor.c
+ * command-line tool for decoding a PT trace, and converting it into an instruction-only
+ * memtrace composed of 'memref_t's
+ * XXX: Currently, it only counts and prints the instruction count in the trace data.
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <inttypes.h>
+#include <errno.h>
+#include <linux/limits.h>
+
+#include "load_elf.h"
+#include "pt_cpu.h"
+#include "intel-pt.h"
+#include "libipt-sb.h"
+
+#define IN
+#define OUT
+#define INOUT
+
+/* DR IntelPT decoder*/
+typedef struct _dript_decoder_t {
+    /* libipt configuration */
+    struct pt_config config;
+
+    /* libipt instruction decoder*/
+    struct pt_insn_decoder *instr_decoder;
+
+    /* The image section cache. */
+    struct pt_image_section_cache *iscache;
+
+    /* The preload image. */
+    struct pt_image *preload_image;
+
+    /* The perf event sideband decoder configuration. */
+    struct pt_sb_pevent_config sb_pevent_conf;
+
+    /* The sideband session. */
+    struct pt_sb_session *sb_session;
+} dript_decoder_t;
+
+/* A collection of options. */
```

Comment:
So, when `print_stats` is set in this struct, `instr_count` will be set in `dript_output_t`? Add that to the documentation of this struct. `This decides which fields of dript_output_t` will be set.